tests/integration/sdk_api/alerts/test_alerts.py

```diff
@@ -0,0 +1,563 @@
+        modified_alert = self._modify_alert_test(
+            project_name, alert1, created_alert["name"], new_event_name
+        )
+
+        # post new event to make sure the modified alert handles it
+        self._post_event(project_name, new_event_name, alert1["entity"]["kind"])
+
+        alert = self._get_alerts(project_name, modified_alert["name"])
+        self._validate_alert(alert, alert_state=alert_constants.AlertActiveState.ACTIVE)
+
+        # reset alert
+        self._reset_alert(project_name, created_alert["name"])
+
+        alert = self._get_alerts(project_name, created_alert["name"])
+        self._validate_alert(
+            alert, alert_state=alert_constants.AlertActiveState.INACTIVE
+        )
+
+        # reset the alert again, and validate that the state is still inactive
+        self._reset_alert(project_name, created_alert["name"])
+
+        alert = self._get_alerts(project_name, created_alert["name"])
+        self._validate_alert(
+            alert, alert_state=alert_constants.AlertActiveState.INACTIVE
+        )
+
+        # delete alert
+        self._delete_alert(project_name, created_alert["name"])
+
+        alerts = self._get_alerts(project_name)
+        assert len(alerts) == 1
+
+        # try to delete invalid alert
+        self._delete_alert(project_name, name="666")
+
+        self._delete_alert(project_name, created_alert2["name"])
+
+        # validate get alerts on empty system after deletes
+        alerts = self._get_alerts(project_name)
+        assert len(alerts) == 0
+
+        mlrun.get_run_db().delete_project(project_name)
+
+    def test_alert_after_project_deletion(self):
+        # this test checks create alert and post event operations after deleting a project and creating it again
+        # with the same alert and event names
+
+        project_name = "my-new-project"
+        event_name = alert_constants.EventKind.DRIFT_DETECTED
+        alert_name = "drift"
+        alert_summary = "Model {{ $project }}/{{ $entity }} is drifting."
+        alert_entity_kind = alert_constants.EventEntityKind.MODEL
+        alert_entity_project = project_name
+
+        mlrun.new_project(alert_entity_project)
+        self._create_alert(
+            alert_entity_project,
+            alert_name,
+            alert_entity_kind,
+            alert_entity_project,
+            alert_summary,
+            event_name,
+        )
+        self._post_event(alert_entity_project, event_name, alert_entity_kind)
+        mlrun.get_run_db().delete_project(alert_entity_project, "cascade")
+        mlrun.new_project(alert_entity_project)
+        self._create_alert(
+            alert_entity_project,
+            alert_name,
+            alert_entity_kind,
+            alert_entity_project,
+            alert_summary,
+            event_name,
+        )
+        self._post_event(alert_entity_project, event_name, alert_entity_kind)
+
+    def _create_alerts_test(self, project_name, alert1, alert2):
```

Comment:
I feel like a lot of the checks here can be done cleaner with a for loop / pytest.parametrize on `(alert_config, expectation)` (See other places in mlrun that use `expectation`)